src/json.c

```diff
@@ -77,7 +83,95 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type) {
   return rv;
 }
 
-int JSON_StoreInDocField(RedisJSON json, JSONType jsonType, struct DocumentField *df) {
+// Uncomment when support for more types is added
+// static int JSON_getInt32(RedisJSON json, int32_t *val) {
+//   long long temp;
+//   int ret = japi->getInt(json, &temp);
+//   *val = (int32_t)temp;
+//   return ret;
+// }
+
+static int JSON_getFloat32(RedisJSON json, float *val) {
+  double temp;
+  int ret = japi->getDouble(json, &temp);
+  if (REDISMODULE_OK != ret) {
+    long long temp2;
+    ret = japi->getInt(json, &temp2);
+    *val = (float)temp2;
+  } else {
+    *val = (float)temp;
+  }
+  return ret;
+}
+
+// Uncomment when support for more types is added
+// static int JSON_getFloat64(RedisJSON json, double *val) {
+//   int ret = japi->getDouble(json, val);
+//   if (REDISMODULE_OK != ret) {
+//     long long temp;
+//     ret = japi->getInt(json, &temp);
+//     *val = (double)temp;
+//   }
+//   return ret;
+// }
+
+int JSON_StoreVectorInDocField(FieldSpec *fs, JSONResultsIterator arrIter, struct DocumentField *df) {
+  VecSimType type;
+  size_t dim;
+  int (*getElement)(RedisJSON, void *);
+
+  switch (fs->vectorOpts.vecSimParams.algo) {
+    case VecSimAlgo_HNSWLIB:
+      type = fs->vectorOpts.vecSimParams.hnswParams.type;
+      dim = fs->vectorOpts.vecSimParams.hnswParams.dim;
+      break;
+    case VecSimAlgo_BF:
+      type = fs->vectorOpts.vecSimParams.bfParams.type;
+      dim = fs->vectorOpts.vecSimParams.bfParams.dim;
+      break;
+    default: return REDISMODULE_ERR;
```

Comment:
better to not have a default so we get compilation warning if/when we add an enum in the future

Reply:
Good point. 
So in order to silence the warning of uninitialized variables I can ether give them default values which can be a waste of time, and behave as a default but not failing on bad value (probably impossible), or using this default which is a dead code.
I don't know which approach is better. what do you think?

Reply:
Even if we use all available switch case enums - the compiler still complains about missing initializations
`may be used uninitialized` (`-Wmaybe-uninitialized`)
So instead of adding unneeded initializations, it seems OK to have a `default` case to avoid such warnings.